docs/articles/documentation/reference/test-api/testcontroller/eval.md

```diff
@@ -27,7 +27,9 @@ test('My Test', async t => {
 });
 ```
 
-> Since the `eval` method returns a value, and not an object, you cannot chain other methods of the test controller after 'eval'.
+> Important! The `eval` method returns a value and not an object. Always place `eval` in a separate call.
```

Comment:
It's not a good idea to say what type an API member returns in a warning note only. 

I'd address this issue by adding types to the signature, like in [TestCafe API](https://devexpress.github.io/testcafe/documentation/reference/testcafe-api/runner/browsers.html): ` → this` for chainable `TestController` methods and ` → [type]` (` → String`, ` → Object`) for the rest of them (including this one).

We can do this for the `TestController` members first, but since the entire test API is not so large, we may want to cover it all.

You can refer to TypeScript [type definition](https://github.com/DevExpress/testcafe/blob/master/ts-defs-src/test-api/test-controller.d.ts) for reference.

And, regarding this topic, we can duplicate this info with a note that this method is unchainable, as you did.